resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -786,6 +790,15 @@ private[spark] class MesosClusterScheduler(
     }
   }
 
+  /**
+   * Check if the task has already been launched or is pending
+   * If neither, the taskId is outdated and should be ignored
+   * This is to avoid scenarios where an outdated status update arrives
+   * after a supervised driver has already been relaunched
+   */
+  private def taskIsOutdated(taskId: String, state: MesosClusterSubmissionState): Boolean =
+    taskId != state.taskId.getValue && !pendingRetryDrivers.contains(state.driverDescription)
```

Comment:
Does this include the case where the task was launched a while ago and has already finished? (Is that even a valid case?) Would this still return "True"?

Reply:
Yup. So in your example say: 
```
- driver task01 running on agent1
- agent1 crashes
- job relaunched on agent2 as task01-retry-1
- job finishes. Latest state.task.Id == task01-retry-1
- agent1 comes back and tries to relaunch task01
- this method would see that task01 =/= task01-retry-1 && 
  there aren't any pendingRetries and would return True
  preventing a zombie job from reviving
```